src/Evolution/Systems/GeneralizedHarmonic/Equations.cpp

```diff
@@ -0,0 +1,255 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GeneralizedHarmonic/Equations.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/System.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+
+namespace GeneralizedHarmonic {
+
+template <size_t Dim>
+void ComputeDuDt<Dim>::apply(
+    const gsl::not_null<tnsr::aa<DataVector, Dim>*> dt_psi,
+    const gsl::not_null<tnsr::aa<DataVector, Dim>*> dt_pi,
+    const gsl::not_null<tnsr::iaa<DataVector, Dim>*> dt_phi,
+    const tnsr::aa<DataVector, Dim>& psi, const tnsr::aa<DataVector, Dim>& pi,
+    const tnsr::iaa<DataVector, Dim>& phi,
+    const tnsr::iaa<DataVector, Dim>& d_psi,
+    const tnsr::iaa<DataVector, Dim>& d_pi,
+    const tnsr::ijaa<DataVector, Dim>& d_phi, const Scalar<DataVector>& gamma0,
+    const Scalar<DataVector>& gamma1, const Scalar<DataVector>& gamma2,
+    const tnsr::a<DataVector, Dim>& gauge_function,
+    const tnsr::ab<DataVector, Dim>& spacetime_deriv_gauge_function,
+    const Scalar<DataVector>& lapse, const tnsr::I<DataVector, Dim>& shift,
+    const tnsr::II<DataVector, Dim>& upper_spatial_metric,
+    const tnsr::AA<DataVector, Dim>& upper_psi,
```

Comment:
Why `upper_psi` instead of `inverse_psi`?

Reply:
I'd prefer inverse_spacetime_metric myself 🙂

Reply:
Are we preferring `inverse_spatial_metric` too?

Reply:
Yes, these should be the same otherwise people will go insane (at the very least I would because I would forget which one is which always, though I'm not sure I'm not already insane)